public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx

```diff
@@ -131,6 +132,10 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
   };
 
   onDiscard = () => {
+    // if this is a newly created panel we want to remove it from the dashboard
+    // todo how do we check it's a new panel?
```

Comment:
not sure you can right now, would have to add some new state to it, I know we have had a "isNew" state before but seems to be gone now. 

Just  make sure we don't save this new state prop / get's removed when you leave panel edit (without discarding)